drivers: udc_dwc2: add Espressif pre-enable and shutdown hooks

sylvioalves opened the pull request:
Ensure the USB clock is enabled before initializing the controller by invoking the pre_enable hook. This avoids initialization failures when the HAL has not yet configured the clock.

Fixes #96679

sylvioalves committed:
drivers: udc_dwc2: add Espressif pre-enable and shutdown hooks

Ensure the USB clock is enabled before initializing the
controller by invoking the pre_enable hook. This avoids
initialization failures when the HAL has not yet configured
the clock.

Adds shutdown hook to allow interrupt and clock deinit.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

jfischer-no commented on drivers/usb/udc/udc_dwc2_vendor_quirks.h:
```diff
@@ -607,24 +607,29 @@ static inline int esp32_usb_otg_init(const struct device *dev,
+		usb_wrap_ll_phy_enable_pad(phy_ctx->wrap_hal.dev, true);
 		LOG_DBG("PHY enabled");
 	} else {
-		usb_wrap_ll_enable_bus_clock(false);
```
Is it intended to never turn off the clock? Maybe you want to use shutdown quirk to disable it.

sylvioalves committed:
drivers: udc_dwc2: increase stack size for ESP32-S3

Increase default UDC DWS stack size for ESP32-S3 in order
to guarantee quirks are executed without issues.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

tmon-nordic commented:
nitpick: the driver is usb_dwc2 not usb_dw2
